extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -402,6 +451,32 @@ export class VisibilityManager {
     };
   }
 
+  /**
+   * Given the vertical dimensions of an element, a scroll position, and the
+   * viewport height, returns an indication of whether the viewport was
+   * above the element, or below the element, or included the element.
+   *
+   * @param {number} elementTop The top of the element's bounding box.
+   * @param {number} elementHeight The height of the element's bounding box.
+   * @param {number} scrollPos The scroll position, e.g. the document y position
+   *                 at the top of the viewport.
+   * @param {number} viewportHeight The height of the viewport.
+   * @return {!RelativePosition}
+   * @private
+   */
+  getRelativePosition_(elementTop, elementHeight, scrollPos, viewportHeight) {
```

Comment:
nit: can we move the helper function to `layout-rect.js`, or you can use this one
https://github.com/ampproject/amphtml/blob/master/src/layout-rect.js#L157-L165

Reply:
Done.
layoutRectsRelativePos() isn't exactly the desired logic for this use case, so I used both that and layoutRectsOverlap():

Case 1: Element (r2) is completely above the top of the viewport (r1).
layoutRectsRelativePos() would correctly return BOTTOM.

Case 2: Element is partially above the top of the viewport, and partially inside the viewport.
layoutRectsRelativePos() would return BOTTOM, when we want INSIDE.

Case 3: Element is completely inside the viewport.
layoutRectsRelativePos() would correctly return INSIDE.

Case 4: Viewport is completely inside the element.
layoutRectsRelativePos() would return TOP, but we want INSIDE.

Case 5: Element is partially below the bottom of the viewport, and partially inside the viewport.
layoutRectsRelativePos() would return TOP, when we want INSIDE.

Case 6: Element is completely below the bottom of the viewport.
layoutRectsRelativePos() would correctly return TOP.

So, I short-circuited cases 2-5 with layoutRectsOverlap().